src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpCharacters.cs

```diff
@@ -198,5 +199,23 @@ public static int IndexOfInvalidFieldValueChar(string s)
 
             return -1;
         }
+
+        // Disallows control characters but allows extended characters > 0x7F
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int IndexOfInvalidFieldValueCharExtended(string s)
```

Comment:
Unrelated to this PR, but are these methods hot?
My gut-feeling means there may be some room for perf-improvements, so is it worth to have a look?

Reply:
It's called multiple times per response and some values like Set-Cookie could be long, so yes, feel free to take a look.

Reply:
I gave this a quick try, see speedups (depending on the actual length) of 5 to 10.
Will create a PR once this PR is in.

Reply:
Wanted to wait for xplat-intrinsics to be there 😃 
--> https://github.com/dotnet/aspnetcore/pull/44041

Reply:
@gfoidl, what generalized IndexOfXx methods could we expose in runtime to avoid the need to custom code something here? Would https://github.com/dotnet/runtime/issues/68328 suffice?

Reply:
Hm, yes I think `IndexOfAny` for > 5 can do the job.
I didn't think about that, as my prototype dates back to June 2021 and then I didn't tidy up my personal todo with the new possibilities.

I have an idea on how that could work generalized and based on a generator to get the best of both worlds (vector supported vs. not supported). A bit later today (have to leave now) I'll sketch out a prototype of this and share my findings / thoughts in https://github.com/dotnet/runtime/issues/68328.